src/component/project/Project/ProjectForm/ProjectForm.tsx

```diff
@@ -0,0 +1,102 @@
+import PermissionButton from '../../../common/PermissionButton/PermissionButton';
+import { ADMIN } from '../../../providers/AccessProvider/permissions';
+import Input from '../../../common/Input/Input';
+import { TextField, Button } from '@material-ui/core';
+import { useStyles } from './ProjectForm.style';
+import React from 'react';
+
+interface IProjectForm {
+    projectId: string;
+    projectName: string;
+    projectDesc: string;
+    setProjectId: React.Dispatch<React.SetStateAction<string>>;
+    setProjectName: React.Dispatch<React.SetStateAction<string>>;
+    setProjectDesc: React.Dispatch<React.SetStateAction<string>>;
+    handleSubmit: (e: any) => void;
+    handleCancel: () => void;
+    errors: { [key: string]: string };
+    submitButtonText: string;
+    clearErrors: () => void;
+    validateIdUniqueness: () => void;
+}
+
+const ProjectForm = ({
+    handleSubmit,
+    handleCancel,
+    projectId,
+    projectName,
+    projectDesc,
+    setProjectId,
+    setProjectName,
+    setProjectDesc,
+    errors,
+    submitButtonText,
+    validateIdUniqueness,
+    clearErrors,
+}: IProjectForm) => {
+    const styles = useStyles();
+
+    return (
+        <form onSubmit={handleSubmit} className={styles.form}>
+            <h3 className={styles.formHeader}>Project Information</h3>
+
+            <div className={styles.container}>
+                <p className={styles.inputDescription}>
+                    What is your project Id?
+                </p>
+                <Input
+                    className={styles.input}
```

Comment:
We need to make sure this field does not allow spaces. Currently we are doing it like this in the old view: 

```
    <TextField
          label="Project Id"
          name="id"
          placeholder="A-unique-key"
          value={project.id}
          error={!!errors.id}
          helperText={errors.id}
          disabled={editMode}
          variant="outlined"
          size="small"
          onBlur={v => validateId(v.target.value)}
          onChange={v =>
              setValue('id', trim(v.target.value))
          }
      />
```